src/main/java/seedu/address/logic/parser/AddVolunteerCommandParser.java

```diff
@@ -0,0 +1,71 @@
+import static seedu.address.logic.parser.CliSyntax.PREFIX_TAG;
+
+import java.util.Set;
+import java.util.stream.Stream;
+
+import seedu.address.logic.commands.AddVolunteerCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.person.Volunteer;
+import seedu.address.model.person.information.Address;
+import seedu.address.model.person.information.Age;
+import seedu.address.model.person.information.Email;
+import seedu.address.model.person.information.Name;
+import seedu.address.model.person.information.Nric;
+import seedu.address.model.person.information.Phone;
+import seedu.address.model.tag.Tag;
+
+/**
+ * Parses input arguments and creates a new AddElderlyCommand object
+ */
+public class AddVolunteerCommandParser implements Parser<AddVolunteerCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the AddElderlyCommand
+     * and returns an AddElderlyCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public AddVolunteerCommand parse(String args) throws ParseException {
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_PHONE,
+                        PREFIX_EMAIL, PREFIX_ADDRESS, PREFIX_NRIC_VOLUNTEER, PREFIX_AGE,
+                        PREFIX_TAG);
+
+        if (!arePrefixesPresent(argMultimap, PREFIX_NAME, PREFIX_ADDRESS,
+                PREFIX_PHONE, PREFIX_EMAIL, PREFIX_NRIC_VOLUNTEER, PREFIX_AGE)
+                || !argMultimap.getPreamble().isEmpty()) {
+            throw new ParseException(String
+                    .format(MESSAGE_INVALID_COMMAND_FORMAT,
+                            AddVolunteerCommand.MESSAGE_USAGE));
+        }
+
+        Name name = ParserUtil.parseName(argMultimap.getValue(PREFIX_NAME).get());
+        Phone phone = ParserUtil.parsePhone(argMultimap.getValue(PREFIX_PHONE).get());
+        Email email = ParserUtil.parseEmail(argMultimap.getValue(PREFIX_EMAIL).get());
+        Address address = ParserUtil.parseAddress(argMultimap.getValue(PREFIX_ADDRESS).get());
+        Nric nric = ParserUtil.parseNric(argMultimap.getValue(PREFIX_NRIC_VOLUNTEER).get());
+        Age age = ParserUtil.parseAge(argMultimap.getValue(PREFIX_AGE).get());
+        Set<Tag> tagList = ParserUtil.parseTags(argMultimap.getAllValues(PREFIX_TAG));
+
+        Volunteer volunteer = new Volunteer(name, phone, email, address, nric, age, tagList);
+
+        return new AddVolunteerCommand(volunteer);
+    }
+
+    /**
+     * Returns true if none of the prefixes contains empty {@code Optional} values in the given
+     * {@code ArgumentMultimap}.
+     */
+    private static boolean arePrefixesPresent(ArgumentMultimap argumentMultimap, Prefix... prefixes) {
+        return Stream.of(prefixes).allMatch(prefix -> argumentMultimap.getValue(prefix).isPresent());
+    }
```

Comment:
There's an existing method that does the same thing in `commons/util`. Can you update this file to use that instead? thanks 👍

Reply:
Oh nice, I'll do that.